go/runfiles/global.go

```diff
@@ -42,6 +50,26 @@ func Env() ([]string, error) {
 	return r.Env(), nil
 }
 
+var legacyExternalGeneratedFile = regexp.MustCompile(`^bazel-out[/][^/]+/bin/external/([^/]+)/`)
+var legacyExternalFile = regexp.MustCompile(`^external/([^/]+)/`)
+
+// CurrentRepository returns the canonical name of the Bazel repository that
+// contains the source file of the caller of this function with skip frames
+// skipped. For example, passing 0 will return the name of the repository that
+// contains the caller of CurrentRepository.
+func CurrentRepository(skip int) string {
```

Comment:
Can you create functions `CallerRepository` and `CurrentRepository`?

Reply:
I did, is this what you expected?